loadgen/helpers.go

```diff
@@ -0,0 +1,34 @@
+package loadgen
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+)
+
+func VisibilityCountIsEventually(
+	ctx context.Context,
+	client client.Client,
+	request *workflowservice.CountWorkflowExecutionsRequest,
+	expectedCount int,
+	waitAtMost time.Duration,
+) error {
+	deadline := time.Now().Add(waitAtMost)
+	for {
+		visibilityCount, err := client.CountWorkflow(ctx, request)
+		if err != nil {
+			return err
+		}
+		if visibilityCount.Count == int64(expectedCount) {
+			return nil
+		}
+		if time.Now().After(deadline) {
```

Comment:
Often in Go you'll see start captured and this expressed as

```go
		if time.Since(start) >= waitAtMost {
```

Some even put it in a loop, e.g.

```go
for start := time.Now(); time.Since(start) < waitAtMost; {
```

(or similar)